src/scenes/ApplicationRouter/NewApplicationModal.tsx

```diff
@@ -114,12 +118,12 @@ const NewApplicationModal = ({ open, onClose, onSave }: NewApplicationModalProps
     }
 
     if (!!projectId) {
-      const result = await ApplicationService.createApplication(projectId);
-      if (result.requestSucceeded && result.data) {
-        onSave(result.data?.id);
+      const newApplicationId = await create(projectId);
```

Comment:
I think this should go through our normal async thunk request flow.

Reply:
I am thinking that for these one-off calls, we can handle the result directly instead of going through a selector. Especially since we are not re-using or re-querying the results. Same for `reload`.

Reply:
Makes sense. It is quite a bit of boilerplate

Reply:
Yeah to add to that, I think it gets a bit tricky with selectors because for use cases like this, we want to make sure we are not looking at "stale" results as well. So I think it will be a lot of extra logic to sort that out. Just my 2 cents!!

Reply:
I think the idea of "stale" results is handled by the request selector taking a request ID. So if it changes, the stale results will go away immediately. But I could be misunderstanding what you mean by stale here. Also, things like "create an application then save data to it" sounds like logic that should exist in the service layer as opposed to a modal.

Another edge case here is that the modal can be closed between the two requests, which I think would break the onSave flow. Probably not something we have to be concerned with here, but generally the ability to unmount a component would be disabled if there are multiple requests that execute in succession.

Either way, what you have here is fine, no changes needed.

Reply:
Good call! I think I will implement a new service call for "creating a project and then an application with it".